programs/drift/src/math/spot_withdraw.rs

```diff
@@ -36,13 +35,24 @@ pub fn calculate_max_borrow_token(
 
 pub fn check_user_exception_to_withdraw_limits(
     spot_market: &SpotMarket,
-    spot_position: Option<&SpotPosition>,
+    user: Option<&User>,
     token_amount_withdrawn: Option<u128>,
 ) -> DriftResult<bool> {
     // allow a smaller user in QUOTE_SPOT_MARKET_INDEX to bypass and withdraw their principal
     let mut valid_user_withdraw = false;
-    if let Some(spot_position) = spot_position {
-        if spot_position.market_index == QUOTE_SPOT_MARKET_INDEX
+    if let Some(user) = user {
+        let spot_position = user.get_spot_position(spot_market.market_index).unwrap();
+        let net_deposits = user
+            .total_deposits
+            .cast::<i128>()?
+            .safe_sub(user.total_withdraws.cast::<i128>()?)?;
+        msg!(
+            "net_deposits={}({}-{})",
+            net_deposits,
+            user.total_deposits,
+            user.total_withdraws
+        );
+        if net_deposits >= 0
             && spot_position.cumulative_deposits >= 0
             && spot_position.balance_type == SpotBalanceType::Deposit
         {
```

Comment:
what is the benefit of
```
                if user_deposit_token_amount.safe_add(token_amount_withdrawn)?
                    < spot_market
                        .withdraw_guard_threshold
                        .cast::<u128>()?
                        .safe_div(10)?
                {
                    valid_user_withdraw = true;
                }```

Reply:
makes exemption only apply for small accounts